exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -215,21 +220,47 @@ func getPromMetricName(metric *otlp.Metric, ns string) string {
 	return sanitize(b.String())
 }
 
-// Simple helper function that takes the <Signature String - *TimeSeries> map
-// and creates a WriteRequest from the struct -- can move to the helper.go file
-func wrapTimeSeries(tsMap map[string]*prompb.TimeSeries) (*prompb.WriteRequest, error) {
+// Helper function that takes the <Signature String - *TimeSeries> map
+// and creates a batch of WriteRequest from the struct
+func wrapAndBatchTimeSeries(tsMap map[string]*prompb.TimeSeries) ([]*prompb.WriteRequest, error) {
 	if len(tsMap) == 0 {
 		return nil, errors.New("invalid tsMap: cannot be empty map")
 	}
+
+	var requests []*prompb.WriteRequest
 	var tsArray []prompb.TimeSeries
+	missingLastBatch := true
+
 	for _, v := range tsMap {
 		tsArray = append(tsArray, *v)
+		missingLastBatch = true
+
+		if len(tsArray)%checkByteSizeInterval == 0 {
+			wrapped := prompb.WriteRequest{
+				Timeseries: tsArray,
+				// Other parameters of the WriteRequest are unnecessary for our Export
+			}
+
+			if b := proto.Size(&wrapped); b >= maxBatchByteSize || len(tsArray) >= maxBatchMetric {
+				requests = append(requests, &wrapped)
+				tsArray = make([]prompb.TimeSeries, 0)
+
+				missingLastBatch = false
+			}
+		}
 	}
-	wrapped := prompb.WriteRequest{
-		Timeseries: tsArray,
-		// Other parameters of the WriteRequest are unnecessary for our Export
+
+	if missingLastBatch {
+		wrapped := prompb.WriteRequest{
+			Timeseries: tsArray,
+			// Other parameters of the WriteRequest are unnecessary for our Export
```

Comment:
Nit comment. 

Why is this? why other parameters are not needed? If you want to make reader to be aware of this fact, do refactor 
```
wrapped := prompb.WriteRequest{
				Timeseries: tsArray,
			}
```

into a separate function, and explain more in depth on why other parameters are not needed. This avoid duplication :-) I can see that this code is copy and pasted from above :-)

Reply:
Yup, thats a good idea! I should have thought of that. I'll add a new function to avoid the duplication.